### PR TITLE
Rework log fields

### PR DESCRIPTION
**Pull request by jack-berg**
Solves several of the issues in #3804. Replaces #3811.

This PR includes a couple of changes to log fields, based on conversations in the 11/4 SIG:
- `setTraceId(..)`, `setSpanId(..)`, and `setFlags(..)` are replaced with `setContext(Context)`
- `getTraceId()`, `getSpanId()` and `getFlags()` are replaced with `SpanContext getSpanContext()`
- `setAttributes(..)` now replaces attributes instead of merging them. This is less surprising.
- `setBody(Body)` is removed. Only `setBody(String)` is currently supported. On the read side `Body getBody()` still returns a `Body`.
- The nullability of fields has been been addressed. Generally, null is avoid if the response has an alternativel
  - `SpanContext getSpanContext()` returns `SpanContext.getInvalid()` if unset.
  - `Severity getSeverity()` returns `Severity#UNDEFINED_SEVERITY_NUMBER` if unset.
  - `String getSeverityText()` is nullable and returns null if unset.
  - `String getName()` is nullable and returns null if unset.
  - `Body getBody()` returns `Body#emptyBody()` if unset.
  - `Attributes getAttributes()` returns `Attributes#empty()` if unset.

### CODE REVIEW

File: sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/Body.java [hunk 1]
```diff
@@ -31,8 +32,18 @@ static Body stringBody(String stringValue) {
     return StringBody.create(stringValue);
   }
 
+  /**
+   * Return an empty {@code Body}.
+   *
+   * @return a {@code Body} without a value.
+   */
+  static Body emptyBody() {
```
Comment (anuraaga) [inline sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/Body.java:40]:
```suggestion
  static Body empty() {
```

Comment (anuraaga) [reply sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/Body.java:40]: Some APIs may repeat the type name in factories for static import but I guess we haven't been

File: sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/EmptyBody.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.logs.data;
+
+class EmptyBody implements Body {
+
+  static final EmptyBody INSTANCE = new EmptyBody();
```
Comment (anuraaga) [inline sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/EmptyBody.java:10]: Can use enum singleton pattern too

**Review by anuraaga — APPROVED** [review]
Looks much nicer thanks!

File: sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/Body.java (hunk 1 above)
Comment (jack-berg) [reply sdk/logs/src/main/java/io/opentelemetry/sdk/logs/data/Body.java:40]: I like this better. I think we can change `stringBody(String)` to just `string(String)` to match this.

**Review by anuraaga — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

**Pull request merged**